app/src/main/java/com/kickstarter/viewmodels/SettingsViewModel.java

```diff
@@ -161,6 +179,26 @@ public ViewModel(final @NonNull Environment environment) {
           this.logout.onNext(null);
         });
 
+      this.optIntoFollowing
+        .compose(bindToLifecycle())
+        .filter(checked -> checked)
+        .subscribe(__ -> this.userInput.onNext(this.userOutput.getValue().toBuilder().social(true).build()));
+
+      this.optIntoFollowing
+        .compose(bindToLifecycle())
+        .filter(checked -> !checked)
+        .subscribe(__ -> this.showConfirmFollowingOptOutPrompt.onNext(null));
+
+      this.optOutOfFollowing
+        .compose(bindToLifecycle())
+        .filter(optOut -> optOut)
+        .subscribe(__ -> this.userInput.onNext(this.userOutput.getValue().toBuilder().social(false).build()));
+
+      this.optOutOfFollowing
+        .compose(bindToLifecycle())
+        .filter(optOut -> !optOut)
```

Comment:
I'm not sure if it might make sense to consolidate these two subscribers into one:

```
.subscribe(optOut -> 
   if (optOut) {
     this.userInput.onNext(this.userOutput.getValue().toBuilder().social(false).build())
   } else {
     this.hideConfirmFollowingOptOutPrompt.onNext(null)
   })
```

Reply:
I'm glad you called this out because I had it that way and changed it. What do you think is best? I was trying to avoid doing an `if/else` because it's not very functional 😭but then this way is not very readable.